src/SonarScanner.MSBuild.Tasks/GetAnalyzerSettings.cs

```diff
@@ -324,33 +334,65 @@ private AnalyzerSettings GetLanguageSpecificSettings(AnalysisConfig config)
             return settings;
         }
 
+        /// <summary>
+        /// Merges and returns the supplied list of analyzer paths. In case of duplicate
+        /// SonarAnalyzers, the path from the sonarAnalyzerPaths list is used.
+        /// </summary>
+        private string[] MergeAnalyzersLists(IEnumerable<string> sonarAnalyzerPaths, IEnumerable<string> userProvidedAnalyzerPaths)
+        {
+            var nonNullSonarAnalyzerPaths = sonarAnalyzerPaths ?? Enumerable.Empty<string>();
```

Comment:
is it truly possible that sonarAnalyzerPaths is null? 

it's only called with `settings.AnalyzerPlugins.SelectMany(ap => ap.AssemblyPaths)` - if `AnalyzerPlugins` is null, it will fail earlier

so you replace this with a Debug.Assert instead - and we get rid of the extra variable